cluster/provider.go

```diff
@@ -8,4 +13,6 @@ type Provider interface {
 	GetAdvertiseAddress() string
 	GetRemoteAddress() string
 	ListenClusterEvents() <-chan struct{}
+	AllocateEndpoint(string, string, *network.EndpointSettings) (string, types.NetworkCreateRequest, error)
```

Comment:
Agree. Now that I published the e2e change in docker/docker, I will fix this sort of shortcuts.